packages/ui/src/components/Visualization/ContextToolbar/FlowType/FlowTypeSelector.test.tsx

```diff
@@ -0,0 +1,217 @@
+import { act, fireEvent, render, waitFor } from '@testing-library/react';
+import { FlowTypeSelector } from './FlowTypeSelector';
+import { EntitiesContext } from '../../../../providers/entities.provider';
+import { sourceSchemaConfig, SourceSchemaType } from '../../../../models/camel';
+import { CamelRouteVisualEntity } from '../../../../models/visualization/flows';
+import { Schema } from '../../../../models';
+import { EntitiesContextResult } from '../../../../hooks';
+
+const config = sourceSchemaConfig;
+config.config[SourceSchemaType.Integration].schema = { schema: { name: 'Integration', description: 'desc' } } as Schema;
+config.config[SourceSchemaType.Pipe].schema = { schema: { name: 'Pipe', description: 'desc' } } as Schema;
+config.config[SourceSchemaType.Kamelet].schema = { schema: { name: 'Kamelet', description: 'desc' } } as Schema;
+config.config[SourceSchemaType.KameletBinding].schema = {
+  name: 'kameletBinding',
+  schema: { description: 'desc' },
+} as Schema;
+config.config[SourceSchemaType.Route].schema = { schema: { name: 'route', description: 'desc' } } as Schema;
+
+let contextValue = {
+  currentSchemaType: SourceSchemaType.Integration,
+  visualEntities: [{ id: 'entity1' } as CamelRouteVisualEntity, { id: 'entity2' } as CamelRouteVisualEntity],
+};
+
+const onSelect = jest.fn();
+const renderWithContext = () => {
+  return render(
+    <EntitiesContext.Provider value={contextValue as unknown as EntitiesContextResult}>
+      <FlowTypeSelector onSelect={onSelect} />
+    </EntitiesContext.Provider>,
+  );
+};
+describe('FlowTypeSelector.tsx', () => {
+  test('component renders', () => {
+    const wrapper = renderWithContext();
```

Comment:
[:nail_care:] What do you think about keeping this abstraction over rendering the component but as a custom component instead.

The usage would be:

```typescript
render(<FlowTypeSelectorWithContext />)
```

And the component would be:
```jsx
const FlowTypeSelectorWithContext: FunctionComponent = () => {
  return (
    <EntitiesContext.Provider value={contextValue as unknown as EntitiesContextResult}>
        <FlowTypeSelector onSelect={onSelect} />
    </EntitiesContext.Provider>
  );
};

Reply:
Yes, mostly readability.